extension/src/asap.rs

```diff
@@ -124,22 +95,15 @@ fn asap_final_inner(
             if !state.sorted {
                 points.sort_by_key(|p| p.ts);
             }
-            // In following the ASAP reference implementation, we only downsample if the number
-            // of points is at least twice the resolution.  Otherwise we keep the number of
-            // points, but still normalize them to equal sized buckets.
-            let downsample_interval = if points.len() >= 2 * state.resolution as usize {
-                find_downsample_interval(&points, state.resolution as i64)
-            } else {
-                (points.last().unwrap().ts - points.first().unwrap().ts) / points.len() as i64
-            };
-            let mut normal = downsample_and_gapfill_to_normal_form(&points, downsample_interval);
+
             let start_ts = points.first().unwrap().ts;
```

Comment:
Can this happen from user error alone?  If so, we ought to provide an error message by changing `.unwrap()` to `.expect("points must not be empty")` or something.  (Same question for another unwrap below in asap_on_timevector)

Is it possible for end_ts to be the same as start_ts and does that matter?

Reply:
This should be impossible to hit during normal use.  We exit this function right away if called on `None` and won't create a `Some` unless we have at least on value.  Similarly we need a time/value pair to create a timevector.  If the data is only one point we should just return that point, and, now that I just fixed the div0 we were hitting in that case, we do so.